zero_bin/common/src/fs.rs

```diff
@@ -17,6 +19,20 @@ pub fn get_previous_proof(path: Option<PathBuf>) -> anyhow::Result<Option<Genera
     let path = path.unwrap();
     let file = File::open(path)?;
     let des = &mut serde_json::Deserializer::from_reader(&file);
-    let proof: GeneratedBlockProof = serde_path_to_error::deserialize(des)?;
-    Ok(Some(proof))
+    let proof: Vec<GeneratedBlockProof> = serde_path_to_error::deserialize(des)?;
+    // Individual proofs are serialized as vector to match other output formats.
+    if proof.len() != 1 {
+        return Err(anyhow!("Invalid proof format."));
+    }
+
+    std::fs::write(
+        "inputs.json",
+        serde_json::to_string(&PublicValues::from_public_inputs(
+            &proof[0].intern.public_inputs,
+        ))
+        .unwrap(),
```

Comment:
Why don't you return error from the function here? By definition we want to bubble up errors and handle them on the application level, not to panic without clue.

Reply:
This is a leftover from debugging something else, shouldn't have been here 😬